Compile error with VK_USE_PLATFORM_XLIB_KHR

It looks like a name collision with `True` and `False` macros from Xlib.h

code:
```
#define VK_USE_PLATFORM_XLIB_KHR
#include <vulkan/vulkan.hpp>
```
clang++ output:
```
In file included from hpptest.cpp:2:
Vulkan-Hpp/vulkan/vulkan.hpp:6923:33: error: expected unqualified-id
  VULKAN_HPP_CONSTEXPR uint32_t False                            = VK_FALSE;
                                ^
/usr/include/X11/Xlib.h:85:15: note: expanded from macro 'False'
#define False 0
              ^
In file included from hpptest.cpp:2:
Vulkan-Hpp/vulkan/vulkan.hpp:6944:33: error: expected unqualified-id
  VULKAN_HPP_CONSTEXPR uint32_t True                             = VK_TRUE;
                                ^
/usr/include/X11/Xlib.h:84:14: note: expanded from macro 'True'
#define True 1
             ^
2 errors generated.
```

--- a/snippets/defines.hpp
+++ b/snippets/defines.hpp
@@ -14,6 +14,17 @@
 #  undef MemoryBarrier
 #endif
 
+// XLib.h defines True/False, which collides with our vk::True/vk::False
+// ->  undef them and provide some namepace-secure constexpr
+#if defined( True )
+#  undef True
+constexpr int True = 1;
+#endif
+#if defined( False )
+#  undef False
+constexpr int False = 0;
+#endif
+
 #if defined(__GNUC__)
 # define GCC_VERSION ( __GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__ )
 #endif

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -140,6 +140,17 @@ static_assert( VK_HEADER_VERSION == 255, "Wrong VK_HEADER_VERSION!" );
 #  undef MemoryBarrier
 #endif
 
+// XLib.h defines True/False, which collides with our vk::True/vk::False
+// ->  undef them and provide some namepace-secure constexpr
+#if defined( True )
+#  undef True
+constexpr int True = 1;
+#endif
+#if defined( False )
+#  undef False
+constexpr int False = 0;
+#endif
+
 #if defined( __GNUC__ )
 #  define GCC_VERSION ( __GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__ )
 #endif

